### PR TITLE
feat: support paste from word and lark to univer doc

### PR DESCRIPTION
**Pull request by Jocs**
- [x] Multi-selection paste, after pasting, the cursor is placed behind the active selection
- [x] Word paste plugin
- [x] lark paste plugin

image

### CODE REVIEW

File: packages/base-docs/src/commands/commands/core-editing.command.ts [hunk 1]
```diff
@@ -103,7 +103,7 @@ export interface IDeleteCommandParams {
 }
 
 /**
- * The command to delete text.
+ * The command to delete text. mainly used in BACKSPACE.
```
Comment (wzhudev) [inline packages/base-docs/src/commands/commands/core-editing.command.ts:106]:
```suggestion
 * The command to delete text, mainly used in BACKSPACE.
```

File: packages/ui-plugin-docs/src/controllers/clipboard.controller.ts [hunk 2]
```diff
@@ -1,20 +1,221 @@
...
             this.disposeWithMe(this._commandService.registerAsMultipleCommand(command))
         );
+        [InnerPasteCommand].forEach((command) => this.disposeWithMe(this._commandService.registerCommand(command)));
+    }
+
+    private commandExecutedListener() {
+        const updateCommandList = [DocCutCommand.id, DocCopyCommand.id, DocPasteCommand.id];
+
+        this.disposeWithMe(
+            this._commandService.onCommandExecuted((command: ICommandInfo) => {
```
Comment (wzhudev) [inline packages/ui-plugin-docs/src/controllers/clipboard.controller.ts:49]: Please check `registerAsMultipleCommand` in `submodules/univer/packages/ui-plugin-sheets/src/controllers/clipboard/clipboard.controller.ts`.

File: packages/ui-plugin-docs/src/controllers/clipboard.controller.ts [hunk 3]
```diff
@@ -1,20 +1,221 @@
...
+
+                    default:
+                        throw new Error(`Unhandled command ${command.id}`);
+                }
+            })
+        );
+    }
+
+    private async handlePaste() {
+        const { _docClipboardService: clipboard } = this;
```
Comment (wzhudev) [inline packages/ui-plugin-docs/src/controllers/clipboard.controller.ts:78]: IMHO, these should be moved to DocPasteCommand.

File: packages/ui-plugin-docs/src/services/clipboard/html-to-udm/converter.ts [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+
+                if (afterProcessRule) {
+                    afterProcessRule.handler(doc, node as HTMLElement);
+                }
+            }
+        }
+    }
+}
+
+export default HtmlToUDMService;
```
Comment (wzhudev) [inline packages/ui-plugin-docs/src/services/clipboard/html-to-udm/converter.ts:111]: No export default please.

File: packages/base-docs/src/doc-plugin.ts [hunk 5]
```diff
@@ -106,7 +106,7 @@ export class DocPlugin extends Plugin {
         // this._markDocAsFocused();
     }
 
-    initializeCommands(): void {
+    private initializeCommands(): void {
```
Comment (wzhudev) [inline packages/base-docs/src/doc-plugin.ts:109]: Please add a prefix underbar to private members of a class.

**Review by wzhudev — CHANGES_REQUESTED** [review]
Some code styles issues need to be discussed.

File: packages/ui-plugin-docs/src/controllers/clipboard.controller.ts (hunk 3 above)
Comment (Jocs) [reply packages/ui-plugin-docs/src/controllers/clipboard.controller.ts:78]: Why need moved this to DocPasteCommand？

File: packages/ui-plugin-docs/src/controllers/clipboard.controller.ts (hunk 2 above)
Comment (Jocs) [reply packages/ui-plugin-docs/src/controllers/clipboard.controller.ts:49]: Do you mean need use `registerAsMultipleCommand ` here?

File: packages/base-docs/src/doc-plugin.ts (hunk 5 above)
Comment (Jocs) [reply packages/base-docs/src/doc-plugin.ts:109]: I've talked to you before about whether it's necessary to add an underline prefix to a private method, so let's add a prefix uniformly

File: packages/ui-plugin-docs/src/controllers/clipboard.controller.ts (hunk 2 above)
Comment (wzhudev) [reply packages/ui-plugin-docs/src/controllers/clipboard.controller.ts:49]: Yes.

Comment (Jocs) [reply packages/ui-plugin-docs/src/controllers/clipboard.controller.ts:49]: 你是指 InnerPasteCommand 为啥没有通过 registerAsMultipleCommand 注册吗？InnerPasteCommand 是一个实际执行 mutation 的 command，感觉没有必要 registerAsMultipleCommand？

**Review by wzhudev — APPROVED** [review]

**Pull request merged**